src/components/List.jsx

```diff
@@ -1,27 +1,76 @@
-import "./List.css";
-
-import "./ListElement.css";
-
-function List({ activities, isGoodWeather, onDelete }) {
-  const goodWeather = "The weather is awesome! Go outside and: ";
-  const badWeather = "Bad weather outside! Here is what you can do now: ";
-
- return (
-    <ul className="activityList">
-      <h2>{isGoodWeather ? goodWeather : badWeather}</h2>
-      {activities.map((activity) => (
-        <li key={activity.id} className="ListElement">
-          {activity.name}
-          <button
-            onClick={() => onDelete(activity.id)}
-            className="ListElement__button"
-          >
-            x
-          </button>
-        </li>
-      ))}
-    </ul>
-  );
-}
+import Form from './Form';
+import ListElement from './ListElement';
+import ErrorComp from './ErrorComp';
+import useLocalStorageState from 'use-local-storage-state';
+import { useEffect, useState } from 'react';
+import { uid } from 'uid';
+import './List.css';
+import './ListElement.css';
+
+const apiURL = 'https://example-apis.vercel.app/api/weathertt';
+
+export default function List() {
+  const [activities, setActivities] = useLocalStorageState('activities', { defaultValue: [] });
+  const [weather, setWeather] = useState();
+
+  function handleAddActivity(newActivity) {
+    // Rename & Destructure
+    const { activityInput: name, checkbox } = newActivity;
+    setActivities([{ id: uid(), name, checkbox }, ...activities]);
+  }
+
+  function handleDeleteActivity(id) {
+    setActivities((prevActivities) => {
+      return prevActivities.filter((activity) => activity.id !== id);
+    });
+  }
 
-export default List;
+  // Fetch API
+  async function getWeather() {
+    try {
+      const response = await fetch(apiURL);
+      const data = await response.json();
+      setWeather(data);
+    } catch (e) {
+      setWeather('fetchError');
+      console.log('Our Error', e);
```

Comment:
```suggestion
    } catch (error) {
      setWeather('fetchError');
      console.log('Our Error', error);
```
weil "e" in verschiedenen Kontexten "error" oder "event" sein könnte, immer gerne ausschreiben, dann gibts keine Verwirrung für niemanden :)